qutip/solver/flimesolve.py

```diff
@@ -0,0 +1,646 @@
+from .solver_base import Solver
+from .result import Result
+from time import time
+from ..ui.progressbar import progress_bars
+from qutip.solver.floquet import fsesolve, FloquetBasis
+
+
+def _floquet_rate_matrix(floquet_basis,
+                         Nt,
+                         T,
+                         c_ops,
+                         c_op_rates,
+                         args,
+                         time_sense=0):
+    '''
+    Parameters
+    ----------
+    floquet_basis : FloquetBasis Object
+        The FloquetBasis object formed from qutip.floquet.FloquetBasis
+    Nt : Int
+        Number of points in one period of the Hamiltonian
+    c_ops : list
+        list of collapse operator matrices as Qobjs
+    c_op_rates : list
+        List of collapse operator rates/magnitudes.
+    args : *dictionary*
+        dictionary of parameters for time-dependent Hamiltonians and
+        collapse operators. ***FIX THIS NEXT BIT*** Currently, the first
+        element MUST be the frequency-dependence of the Hamiltonian
+    time_sense : float 0-1,optional
+        the time sensitivity or secular approximation restriction of 
+        FLiMESolve. Decides "acceptable" values of (frequency/rate) for rate 
+        matrix entries. Lower values imply rate occurs much faster than
+        rotation frtequency, i.e. more important matrix entries. Higher
+        values meaning rates cause changes slower than the Hamiltonian rotates,
+        i.e. the changes average out on longer time scales.
+
+    Returns
+    -------
+    total_R_tensor : 2D Numpy matrix
+        This is the 2D rate matrix for the system,created by summing the
+            Rate matrix of each individual collapse operator. Something
+            something Rate Matrix something something linear Operator.
+    '''
+    Hdim = len(floquet_basis.e_quasi)  # Dimensionality of the Hamiltonian
+    args_key = list(args.values())[0]  # Frequency dependence of Hamiltonian
+
+    # Forming tlist to take FFT of collapse operators
+    time = T
+    dt = time / Nt
+    tlist = np.linspace(0, time - dt, Nt)
+
+    total_R_tensor = {}
+    for cdx, c_op in enumerate(c_ops):
+        '''
+        Transforming the lowering operator into the Floquet Basis
+            and taking the FFT
+        '''
+        c_op_Floquet_basis = np.array(
+            [floquet_basis.to_floquet_basis(c_op, t).full() for t in tlist])
+
+        c_op_Fourier_amplitudes_list = np.array(
+            np.fft.fft(c_op_Floquet_basis, axis=0) / len(tlist)
+        )
+
+        '''
+        Next step is to form arrays of the rate products (rate_products)
+            and the frequency at which these components rotate (delta_array),
+            then dividing the arrays to form the rate quotient array.
+        
+        In quotient array, setting any undefined value, i.e. those for which
+            rate_products = 0, equal to time_sense+1 so that the mask
+            arrays mask out these values.
+        '''
```

Comment:
```suggestion
        # Next step is to form arrays of the rate products (rate_products)
        # and the frequency at which these components rotate (delta_array),
        # then dividing the arrays to form the rate quotient array.
        
        # In quotient array, setting any undefined value, i.e. those for which
        # rate_products = 0, equal to time_sense+1 so that the mask
        # arrays mask out these values.
```